src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java

```diff
@@ -196,4 +206,56 @@ void should_create_new_token_when_token_removed_from_cache() {
         verify(users, times(2)).getUser(any());
         verify(idamClient, times(2)).authenticateUser(any(), any());
     }
+
+
+    @Test
+    public void should_get_userDetail_when_no_error() {
+        given(idamClient.getUserDetails(JWT)).willReturn(USER_DETAILS);
+
+        UserDetails userDetails1 =
+            idamCachedClient.getUserDetails(JWT);
+
+        assertThat(userDetails1).usingRecursiveComparison().isEqualTo(USER_DETAILS);
+        verify(idamClient).getUserDetails(any());
+    }
+
+    @Test
+    public void should_retrieve_userDetail_from_cache_when_value_in_cache() {
+        given(idamClient.getUserDetails(JWT)).willReturn(USER_DETAILS);
+
+        UserDetails userDetails1 =
+            idamCachedClient.getUserDetails(JWT);
+        UserDetails userDetails2 =
+            idamCachedClient.getUserDetails(JWT);
+
+        assertThat(userDetails1).usingRecursiveComparison().isEqualTo(USER_DETAILS);
+        assertThat(userDetails2).usingRecursiveComparison().isEqualTo(userDetails1);
+        verify(idamClient).getUserDetails(any());
+    }
+
+    @Test
+    public void should_invalidate_userDetail_when_cached_token_removed_from_cache() {
+        String jurisdiction1 = "divorce";
+
+        given(users.getUser(jurisdiction1)).willReturn(new Credential(USERNAME, PASSWORD));
+        given(idamClient.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT, JWT2);
+
+        UserDetails expectedUserDetails1 =  new UserDetails("12","q@a.com","","",null);
+
+        given(idamClient.getUserDetails(JWT)).willReturn(expectedUserDetails1, USER_DETAILS);
```

Comment:
Suggestion: consider creating another var, called `expectedUserDetails2` or `expectedUserDetailsAfter...`, even if the object behind it is USER_DETAILS. It will make this test easier to follow.